system/Database/OCI8/Connection.php

```diff
@@ -120,13 +120,13 @@ private function isValidDSN(): bool
      */
     public function connect(bool $persistent = false)
     {
-        if (empty($this->DSN) && ! $this->isValidDSN()) {
+        if ($this->DSN === null || $this->DSN === '' || ! $this->isValidDSN()) {
```

Comment:
Wouldn't it make more sense to move these checks to the `isValidDSN()` method?

Reply:
Exactly!